pkg/cli/cmds/agent.go

```diff
@@ -80,5 +80,10 @@ func agentSubcommands() cli.Commands {
 func AgentRun(clx *cli.Context) error {
 	validateCloudProviderName(clx)
 	validateProfile(clx, "agent")
+	stop := make(chan struct{})
+	if err := startService(stop); err != nil {
+		return err
+	}
+	go watchService(stop)
```

Comment:
Can you move all these lines into a function that's a noop on Linux, rather than unconditionally creating a channel that Linux doesn't even use?

Reply:
kept them out here so it would make more sense that the cmd level is where a stop channel is created and acknowledged instead of deep in a library called "windows". i can tho, it's not hard to move

Reply:
but Linux doesn't need the stop channel - it only exists to be passed into a goroutine that is a noop on Linux. I say move all of it into the windows package.

Reply:
sure, can do